### PR TITLE
Document how to create a common baseline file for multi module gradle projects

### LINKED ISSUE #2100: DetektAll with baseline fails with error
I have a multi-module project and I want to create one baseline file for the entire project instead of one per module, so I'm trying to replicate the detektAll task that is created in this project's build.gradle.kts file. However, I have had to translate this task into Groovy, and it looks like this:
```
task detektAll(type: io.gitlab.arturbosch.detekt.Detekt) {
    description = 'Runs over whole code base without the starting overhead for each module.'
    config = files("$rootDir/detekt-config.yml")
    parallel = true
    input = files(projectDir)
    include '**/*.kt'
    include '**/*.kts'
    exclude '**/resources/**'
    exclude '**/build/**'
    baseline = file("$rootDir/config/detekt/baseline.xml")
    reports {
        xml {
            enabled = true
        }
        html {
            enabled = true
        }
        txt {
            enabled = true
        }
    }
}
```
However, running this fails with the error  
```
A problem was found with the configuration of task ':detektAll'.
File (my root directory)/config/detekt/baseline.xml does not exist.
```
Is there a fix for this?

### PR DESCRIPTION
**Pull request by arturbosch**
Closes #2100

### CODE REVIEW

File: docs/pages/baseline.md [hunk 1]
```diff
@@ -31,7 +31,38 @@ the `--report txt:path/to/report` cli flag.
...
+    }
+}
+
+task projectBaseline(type: io.gitlab.arturbosch.detekt.DetektCreateBaselineTask) {
+    description = "Overrides current baseline."
+
+    input = files(rootProject.projectDir)
+    config = files("$rootDir/detekt/config.yml")
+    baseline = file(new File(rootProject.projectDir, "/detekt/baseline.xml"))
+    failFast = true
```
Comment (adrianczuczka) [inline docs/pages/baseline.md]: As mentioned in https://github.com/arturbosch/detekt/issues/2100, `failFast = true` this caused my baseline task to fail. I would replace this with `ignoreFailures = true`.

File: docs/pages/baseline.md [hunk 2]
```diff
@@ -31,7 +31,38 @@ the `--report txt:path/to/report` cli flag.
...
+    parallel = true
+    include '**/*.kt'
+    include '**/*.kts'
+    exclude '**/test/resources/**'
+}
+```
```
Comment (adrianczuczka) [inline docs/pages/baseline.md]: Can we move the entire task to the `subprojects` block? It did not work correctly for me without doing that.

File: docs/pages/baseline.md [hunk 3]
```diff
@@ -31,7 +31,38 @@ the `--report txt:path/to/report` cli flag.
...
+        ...
+    }
+}
+
+task projectBaseline(type: io.gitlab.arturbosch.detekt.DetektCreateBaselineTask) {
+    description = "Overrides current baseline."
+
+    input = files(rootProject.projectDir)
+    config = files("$rootDir/detekt/config.yml")
+    baseline = file(new File(rootProject.projectDir, "/detekt/baseline.xml"))
```
Comment (adrianczuczka) [inline docs/pages/baseline.md]: Can we replace this with `baseline = file("${rootProject.projectDir}/detekt-baseline.xml")`? As far as I know it does the same thing and as mentioned in https://github.com/arturbosch/detekt/issues/2100, there is a weird IDE error on my laptop (which does not cause the build to fail, however) related to `new File`. Also, we use the suggested syntax for both `input` and `config`.

**Comment by arturbosch** [comment]
@adrianczuczka thanks for the suggestions, I've incorporated them and added a `detektProjectBaseline` task for detekt too.

File: docs/pages/baseline.md (hunk 2 above)
Comment (arturbosch) [reply docs/pages/baseline.md]: It should work on top level like in our build file. Inside `subprojects` every sub-project would get a baseline task hm :/

**Pull request merged**